Signal/src/AppDelegate.m

```diff
@@ -596,6 +597,9 @@ - (void)handleActivation
                 // Mark all "attempting out" messages as "unsent", i.e. any messages that were not successfully
                 // sent before the app exited should be marked as failures.
                 [[[OWSFailedMessagesJob alloc] initWithPrimaryStorage:[OWSPrimaryStorage sharedManager]] run];
+                // Mark all "incomplete" calls as missed, e.g. any incoming or outgoing calls that were not
+                // connected, failed or hung up before the app existed should be marked as missed.
+                [[[OWSIncompleteCallsJob alloc] initWithPrimaryStorage:[OWSPrimaryStorage sharedManager]] run];
```

Comment:
In call records, we use the "incomplete" state to distinguish "calls which are in progress but haven't yet connected" from the "calls which completed/failed/were missed."  

* We had a "missed" state for incoming calls but not for outgoing calls.  We just left them as "incomplete" forever.  We need to distinguish these states.  For example, now that we can leave ongoing calls and review the conversation history, we want "ongoing" calls to show up differently than "missed" calls.  They should have different copy and shouldn't have a "call again/call back" button.  So I've added a "missed" state for outgoing calls.
* We use `OWSFailedMessagesJob` to ensure that "sending" messages are marked as failed on startup.  We need to do something analogous for calls, marking "incomplete" calls as "missed."
* We might want to rename "incomplete" to "connecting or ringing".

Reply:
👍